libgoal/transactions.go

```diff
@@ -191,8 +194,86 @@ func (c *Client) SignAndBroadcastTransaction(walletHandle, pw []byte, utx transa
 	return c.BroadcastTransaction(stx)
 }
 
+// generateRegistrationTransaction returns a transaction object for registering a Participation with its parent this is
+// similar to account.Participation.GenerateRegistrationTransaction.
+func generateRegistrationTransaction(part generated.ParticipationKey, fee basics.MicroAlgos, txnFirstValid, txnLastValid basics.Round, leaseBytes [32]byte) (transactions.Transaction, error) {
+	addr, err := basics.UnmarshalChecksumAddress(part.Address)
+	if err != nil {
+		return transactions.Transaction{}, err
+	}
+
+	if len(part.Key.VoteParticipationKey) != 32 {
+		return transactions.Transaction{}, fmt.Errorf("voting key is the wrong size, should be 32 but it is %d", len(part.Key.VoteParticipationKey))
+	}
+
+	var votePk [32]byte
+	copy(votePk[:], part.Key.VoteParticipationKey[:])
+
+	if len(part.Key.SelectionParticipationKey) != 32 {
+		return transactions.Transaction{}, fmt.Errorf("selection key is the wrong size, should be 32 but it is %d", len(part.Key.VoteParticipationKey))
+	}
+
+	var selectionPk [32]byte
+	copy(selectionPk[:], part.Key.SelectionParticipationKey[:])
+
+	if part.Key.StateProofKey == nil {
+		return transactions.Transaction{}, fmt.Errorf("state proof key pointer is nil")
+	}
+
+	if len(*part.Key.StateProofKey) != len(merklesignature.Verifier{}) {
+		return transactions.Transaction{}, fmt.Errorf("state proof key is the wrong size, should be %d but it is %d", len(merklesignature.Verifier{}), len(*part.Key.StateProofKey))
+	}
+
+	var stateProofPk merklesignature.Verifier
+	copy(stateProofPk[:], (*part.Key.StateProofKey)[:])
+
+	t := transactions.Transaction{
+		Type: protocol.KeyRegistrationTx,
+		Header: transactions.Header{
+			Sender:     addr,
+			Fee:        fee,
+			FirstValid: txnFirstValid,
+			LastValid:  txnLastValid,
+			Lease:      leaseBytes,
+		},
+		KeyregTxnFields: transactions.KeyregTxnFields{
+			VotePK:       votePk,
+			SelectionPK:  selectionPk,
+			StateProofPK: stateProofPk,
```

Comment:
the original function has
```
	if cert := part.StateProofSigner(); cert != nil {
		if includeStateProofKeys { // TODO: remove this check and parameter after the network had enough time to upgrade
			t.KeyregTxnFields.StateProofPK = *(cert.GetVerifier())
		}
	}
```
Are we assuming we waited enough and all participants have falkon keys?

Reply:
Looking through the function I think this has to be the case.  It is called in the `MakeUnsignedGoOnlineTx` transaction, so it would make sense that we are forcing people to have valid keys to go online.  This would also explain why we use this instead of `GenerateRegistrationTransaction()` from `Participation`.  Again that's what my intuition tells me, @winder thoughts?

Reply:
Yes, we now require state proof keys.

Reply:
should we remove this flag from `part.GenerateRegistrationTransaction` as well? I think it means that algokey supports the creation of participation keys without stateproof keys (should that be removed?)